pkg/cmd/dcos.go

```diff
@@ -150,36 +152,21 @@ func updateCorePlugin(ctx api.Context) error {
 
 // invokePlugin calls the binary of a plugin, passing in the arguments it's been given.
 func invokePlugin(ctx api.Context, cmd plugin.Command, args []string) error {
-	executablePath, err := os.Executable()
-	if err != nil {
-		return err
-	}
 	execCmd := exec.Command(cmd.Path, args...)
 	execCmd.Stdout = ctx.Out()
 	execCmd.Stderr = ctx.ErrOut()
 	execCmd.Stdin = ctx.Input()
 
-	execCmd.Env = append(os.Environ(), "DCOS_CLI_EXECUTABLE_PATH="+executablePath)
-
-	switch ctx.Logger().Level {
-	case logrus.DebugLevel:
-		execCmd.Env = append(execCmd.Env, "DCOS_VERBOSITY=2", "DCOS_LOG_LEVEL=debug")
-	case logrus.InfoLevel:
-		execCmd.Env = append(execCmd.Env, "DCOS_VERBOSITY=1", "DCOS_LOG_LEVEL=info")
+	executablePath, err := os.Executable()
+	if err != nil {
+		return err
 	}
-
-	// Pass cluster specific env variables when a cluster is attached.
-	if cluster, err := ctx.Cluster(); err == nil {
-		execCmd.Env = append(execCmd.Env, "DCOS_URL="+cluster.URL())
-		execCmd.Env = append(execCmd.Env, "DCOS_ACS_TOKEN="+cluster.ACSToken())
-
-		insecure := cluster.TLS().Insecure || strings.HasPrefix(cluster.URL(), "http://")
-		if insecure {
-			execCmd.Env = append(execCmd.Env, "DCOS_TLS_INSECURE=1")
-		} else if cluster.TLS().RootCAsPath != "" {
-			execCmd.Env = append(execCmd.Env, "DCOS_TLS_CA_PATH="+cluster.TLS().RootCAsPath)
-		}
+	cluster, err := ctx.Cluster()
+	if err != nil && err != config.ErrNotAttached {
```

Comment:
Previously we ignored this error? Why you changed this behavior?

Reply:
This shouldn't happen in practice but I think it's a valid error to catch.